compat/src/render.js

```diff
@@ -159,6 +159,14 @@ options.vnode = vnode => {
 			});
 		}
 
+		// Adding support for defaultValue in select tag
+		if (type == 'select' && normalizedProps.defaultValue) {
```

Comment:
Let's enhance this check to `normalizeProps.defaultValue != null`. This will ensure we still apply this normalization even if users pass in `0` 👍

Reply:
Right now I have converted it to single select but multi select has issues. When I tried the following code

```javascript
		// Adding support for defaultValue in select tag
		if (type == 'select' && normalizedProps.defaultValue != null) {
			normalizedProps.value = toChildArray(props.children).forEach(child => {
				if (normalizedProps.multiple) {
					child.props.selected =
						normalizedProps.defaultValue.indexOf(child.props.value) != -1;
				} else {
					child.props.selected =
						normalizedProps.defaultValue == child.props.value;
				}
			});
		}
```

for multi selection only the last value is selected. any thoughts on why this is happening ?

Reply:
but I tried logging the value in the loop it was selected during iteration but some got overwritten during the next iteration.

Reply:
oops, I meant to comment under the other comment https://github.com/preactjs/preact/pull/2848#discussion_r530210401 did it here by mistake.